tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py

```diff
@@ -13,24 +15,24 @@ def __init__(self,
                  group_key: str,
                  recommendations_col: str,
                  item_type: str,
-                 module_name: str,
-                 module_version: str):
+                 module_name: str):
         super(GenericRecommendationFormatterTransformer, self).__init__()
         self.group_key = group_key
         self.recommendations_col = recommendations_col
-        self.item_type = item_type
+        self.item_type = item_type.upper()
         self.module_name = module_name
-        self.module_version = module_version
+        assert self.item_type in [x.upper() for x in [c.ALBUM, c.ARTIST, c.PLAYLIST, c.TRACK, c.MIX, c.RADIO,
+                                                      "ARTIST_TOP_TRACKS", "MASTER_BUNDLE_ID", "TRACK_GROUP"]]
 
     def _transform(self, dataset: DataFrame) -> DataFrame:
         return (dataset
                 .withColumn(c.TYPE, F.lit(self.item_type))
-                .withColumn(c.MODULE_ID, F.lit(f"{self.module_name}_v{self.module_version}"))
-                .select(F.col(self.group_key).alias(c.HASH_KEY), c.TYPE, c.MODULE_ID,
+                .withColumn(c.MODULE_ID, F.lit(self.module_name))
+                .select(F.col(self.group_key).cast(T.StringType()).alias(c.HASH_KEY), c.TYPE, c.MODULE_ID,
```

Comment:
all ids should be string

Reply:
Thanks, I was casting to string on my side of the pipeline, I'll remove it :)